ci_framework/roles/openshift_provisioner_node/tasks/main.yml

```diff
@@ -0,0 +1,74 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Create the artifacts directory
+  ansible.builtin.file:
+    path: "{{ cifmw_opn_artifacts_dir }}"
+    state: directory
+  delegate_to: localhost
+
+- name: Add users
+  ansible.builtin.import_tasks:
+    file: add_user.yml
+
+- name: Install required packages for the installer
+  ansible.builtin.import_tasks:
+    file: install_packages.yml
+
+- name: Check network settings are inline with installer requirement
+  ansible.builtin.import_tasks:
+    file: add_bridges.yml
+
+- name: Check required virtualization networks
+  ansible.builtin.include_tasks:
+    file: add_virtual_network.yml
+  loop:
+    - "{{ cifmw_opn_external_bridge_name }}"
+    - "{{ cifmw_opn_prov_bridge_name }}"
+
+- name: Cache the MAC addresses
+  ansible.builtin.set_fact:
+    cifmw_opn_bootstrap_boot_mac: "{{ '52:54:00' | community.general.random_mac }}"
+
+- name: Install and configure DHCP service
+  when:
+    - cifmw_opn_dhcp is defined
+    - cifmw_opn_dhcp | bool
+  vars:
+    data:
+      domain: "{{ cifmw_opn_network.domain | default(ansible_facts.domain) }}"
+      nameservers: "{{ cifmw_opn_network.nameservers | default(ansible_facts.dns.nameservers | join(',')) }}"
+      network: "{{ cifmw_opn_network.network | default(ansible_default_ipv4.network) }}"
+      netmask: "{{ cifmw_opn_network.netmask | default(ansible_default_ipv4.netmask) }}"
+      broadcast: "{{ cifmw_opn_network.broadcast | default(ansible_default_ipv4.broadcast) }}"
+      gateway: "{{ cifmw_opn_network.gateway | default(ansible_default_ipv4.gateway) }}"
+    mac_address: "{{ cifmw_opn_bootstrap_boot_mac }}"
+    fixed_ipv4: "{{ cifmw_opn_bootstrap_ipv4 }}"
+  ansible.builtin.import_tasks:
+    file: config_dhcpd.yml
```

Comment:
I believe setting a DHCP service without limiting it, will create a packet race between the infra dhcp and this one, which can lead to network problems in the network segment.

Reply:
Noted. I have removed the steps and understand why we shouldn't. Fixed in the next patch set.